src/include/uapi/ipc/header.h

```diff
@@ -184,6 +185,14 @@ struct sof_ipc_compound_hdr {
 	uint32_t count;		/**< count of 0 means end of compound sequence */
 } __attribute__((packed));
 
+struct sof_ipc_large_hdr {
```

Comment:
I believe that we planned to follow existing audio fw flow 1:1 (in terms of fragmentation handling and packing individual parameters into tlv arrays inside the payload) in order to translate easily in other OSes in future. The easiest way is to structure the data in the SOF-way with regards to the syntax however would be good to keep the same semantics of the data fields as in the existing design. I am attaching some structure proposal&doc soon.

Reply:
@mmaka1 do you have an ETA for proposal ?

Reply:
@lgirdwood @juimonen I am about to submit the documentation to existing component.h first, followed by the extensions being discussed.

Reply:
The proposal dox content are being added on top of the #953.

Reply:
@juimonen @lrgirdwo Pls take a look at #956. It is 1:1 reflected existing audio FW api wrt semantics and the flow for the driver/application - fw component levels of communication.
Notes:
1. Slight naming changes applied to better explain driver vs. application level parameters.
2. Packed into existing IPC structure, everything except for the SOF specific header and component addressing, is binary exact with the existing audio FW which enables trivial repack in other drivers if needed.